src/main/java/org/generationcp/middleware/dao/GermplasmStudySourceDAO.java

```diff
@@ -52,7 +56,16 @@
 		+ "LEFT JOIN ims_lot lot ON lot.eid = gss.gid \n"
 		+ "WHERE gss.project_id = :studyId ";
 
-	public List<GermplasmStudySourceDto> getGermplasmStudySourceList(final GermplasmStudySourceSearchRequest germplasmStudySourceSearchRequest) {
+	public Map<Integer, GermplasmStudySource> getGermplasmStudySourcesMap(final Set<Integer> gids) {
+		final Criteria criteria = this.getSession().createCriteria(this.getPersistentClass());
+		criteria.createAlias("germplasm", "germplasm");
+		criteria.add(Restrictions.in("germplasm.gid", gids));
+		final List<GermplasmStudySource> result = criteria.list();
+		return result.stream().collect(Collectors.toMap(a -> a.getGermplasm().getGid(), Function.identity()));
```

Comment:
For me, it's better to build the map by GID in the service instead in DAO class so you can reuse for other purposes.
As such, you can rename to public `LIst<GermplasmStudySource> getByGids(final Set<Integer gids>)`

Reply:
Done